src/shared/Grid.jsx

```diff
@@ -1,10 +1,43 @@
 import { useLayoutEffect, useRef } from "react";
-import gridStyle from "./Grid.module.scss";
-
+import gridStyle from './Grid.module.scss';
+import { SelectInput } from '../shared/SelectInput/SelectInput';
+import FormField from '../shared/Form/FormField';
 import GridText from "./GridText";
+import Tooltip from "./ToolTip";
 
 const columnTypeComponents = {
   text: GridText,
+  select: ({ column, row, style }) => (
+    <td style={style}>
+      <SelectInput
+        label=""
+        options={row[column.key].options}
+        value={row[column.key].value}
+        onChange={(e) => row[column.key].onChange(e.target.value)}
+        style={{ width: '100%', margin: 0 }}
+      />
+    </td>
+  ),
+  input: ({ column, row, style }) => (
+    <td style={style}>
+      <FormField
+        label=""
+        type="text"
+        value={row[column.key].value}
+        onChange={(e) => row[column.key].onChange(e.target.value)}
+        style={{ width: '92%',  margin: 0 }}
+      />
+    </td>
+  ),
+  checkbox: ({ column, row, style }) => (
+    <td style={style}>
+      <input
+        type="checkbox"
+        checked={row[column.key].checked}
+        onChange={(e) => row[column.key].onChange(e.target.checked)}
+      />
+    </td>
+  ),
```

Comment:
This approach, where the row contains configuration objects for each field, is interesting, though it's different from what I did in the My Projects interface. Let's discuss this offline and decide how we're going to handle editable fields in the grid so that we can use the same approach everywhere.